doc/sphinx/language/extensions/match.rst

```diff
@@ -244,6 +244,33 @@ Printing of the elimination predicate
    the result type is not printed when Coq knows that it can re-
    synthesize it.
 
+Printing of hidden subterms
++++++++++++++++++++++++++++
+
+.. flag:: Printing Match All Subterms
+
+   In order to be able to cheaply reconstruct the types of the
+   variables bound by `in` and `as`, `match` terms contain the
+   polymorphic universe instance and the parameters of the inductive
+   which is being matched. When this flag is on (it is off by
+   default), this information is displayed as a volatile cast around
+   the match discriminee.
+
+   When the match relies on :flag:`Definitional UIP` the indices are also subterms of the `match` term and are displayed when this flag is on.
+   Otherwise they are not subterms and are displaye as holes `_` when this flag is on.
+
+   .. coqtop:: all
+
+      Set Printing Match All Subterms.
+      Set Printing Universes.
+
+      Polymorphic Inductive eqT@{u} {A:Type@{u}} (a:A) : A -> Type@{u} := reflT : eqT a a.
+      Print eqT_rect.
+
+      Set Definitional UIP.
+      Inductive seq {A} (a:A) : A -> SProp := srefl : seq a a.
+      Print seq_rect.
```

Comment:
```suggestion
   .. coqtop:: all
   
      Print eqT_rect.

   .. coqtop:: in
   
      Set Definitional UIP.
      Inductive seq {A} (a:A) : A -> SProp := srefl : seq a a.

   .. coqtop:: all
      
      Print seq_rect.
```
Omit uninteresting output to make the example shorter.

I expected to see a volatile cast in the Print, i.e. `@term10 :> @type`.  Or am I misunderstanding?

The example would be a lot better if included both Prints with and without `Printing Match All Subterms` so readers can see the differences.

Reply:
>I expected to see a volatile cast in the Print, i.e. @term10 :> @type. Or am I misunderstanding?

![Screenshot_20230120_204140](https://user-images.githubusercontent.com/2461932/213791642-d61fc7c0-2f72-421d-96b0-7a35239bbfbf.png)

Reply:
👍